src/VirtoCommerce.CatalogModule.Web/Controllers/Api/CatalogModuleDynamicAssociationsController.cs

```diff
@@ -42,7 +46,7 @@ public async Task<ActionResult<DynamicAssociationSearchResult>> SearchAssociatio
         [Authorize(ModuleConstants.Security.Permissions.Read)]
         public async Task<ActionResult<DynamicAssociation>> GetAssociationById(string id)
         {
-            var result = await _service.GetByIdsAsync(new []{id});
+            var result = await _dynamicAssociationService.GetByIdsAsync(new []{id});
```

Comment:
Need to add authorization here - check the result in the auth handler.
Also please change Ok(result) so it will return the first found entity